sdk/tools/azure-openrewrite/src/main/java/com/azure/openrewrite/core/http/HttpHeadersCustomRecipe.java

```diff
@@ -0,0 +1,146 @@
+                        .imports("io.clientcore.core.http.models.HttpHeaderName")
+                        .build();
+                    visitedMethodInvocation = replacementTemplate.apply(updateCursor(visitedMethodInvocation), visitedMethodInvocation.getCoordinates().replaceMethod(), visitedMethodInvocation.getArguments().toArray());
+                    maybeAddImport("io.clientcore.core.http.models.HttpHeaderName");
+                }
+
+                methodMatcher = new MethodMatcher("com.azure.core.http.HttpHeaders put(java.lang.String, java.lang.String)");
+                if (methodMatcher.matches(visitedMethodInvocation, true)) {
+                    replacementTemplate = templateBuilder.getJavaTemplateBuilder("set(HttpHeaderName.fromString(#{any(java.lang.String)}), #{any(java.lang.String)})")
+                        .imports("io.clientcore.core.http.models.HttpHeaderName")
+                        .build();
+                    visitedMethodInvocation = replacementTemplate.apply(updateCursor(visitedMethodInvocation), visitedMethodInvocation.getCoordinates().replaceMethod(), visitedMethodInvocation.getArguments().toArray());
+                    maybeAddImport("io.clientcore.core.http.models.HttpHeaderName");
+                }
+
+                methodMatcher = new MethodMatcher("com.azure.core.http.HttpHeaders set(java.lang.String, java.lang.String)");
+                if (methodMatcher.matches(visitedMethodInvocation, true)) {
+                    replacementTemplate = templateBuilder.getJavaTemplateBuilder("set(HttpHeaderName.fromString(#{any(java.lang.String)}), #{any(java.lang.String)})")
+                        .imports("io.clientcore.core.http.models.HttpHeaderName")
+                        .build();
+                    visitedMethodInvocation = replacementTemplate.apply(updateCursor(visitedMethodInvocation), visitedMethodInvocation.getCoordinates().replaceMethod(), visitedMethodInvocation.getArguments().toArray());
+                    maybeAddImport("io.clientcore.core.http.models.HttpHeaderName");
+                }
+
+                methodMatcher = new MethodMatcher("com.azure.core.http.HttpHeaders setAll(java.util.Map)");
+                if (methodMatcher.matches(visitedMethodInvocation, true)) {
+                    replacementTemplate = templateBuilder.getJavaTemplateBuilder("setAll(#{any(io.clientcore.core.http.models.HttpHeaders)})")
+                        .imports("io.clientcore.core.http.models.HttpHeaders")
+                        .build();
+                    visitedMethodInvocation = replacementTemplate.apply(updateCursor(visitedMethodInvocation), visitedMethodInvocation.getCoordinates().replaceMethod(), visitedMethodInvocation.getArguments().toArray());
+                    maybeAddImport("io.clientcore.core.http.models.HttpHeaders");
+                }
```

Comment:
It's not clear how this recipe transforms `Map` into `HttpHeaders`. Where do we loop through all the entries in the Map to create HttpHeaders?

Reply:
This is a bug, I'll track this here: https://github.com/Azure/azure-sdk-for-java/issues/45483